app/controllers/api/v1/shoutouts_controller.rb

```diff
@@ -0,0 +1,62 @@
+require 'digest'
+
+class Api::V1::ShoutoutsController < ApplicationController
+  include ApplicationHelper
+  before_action :require_user!
+
+  ERROR_1 = 'Content shout out is exist!'.freeze
+
+  def create
+    @shoutout = Shoutout.new(shoutout_params)
+
+    return unless @shoutout[:user_id] == current_user.id
+
+    if Shoutout.exists?(digest: shoutout_params[:digest])
+      render json: { error: ERROR_1 }, status: :unprocessable_entity
+
+    elsif @shoutout.save
+      RecipientShoutout.create(records_recipients(@shoutout)) if @shoutout[:recipients].length.positive?
+
+      render json: @shoutout, status: :ok
+    else
+      render json: { error: @shoutout[:errors] }, status: :unprocessable_entity
+    end
+end
+
+  def update
+
+    @shoutout = Shoutout.find(params[:id])
+
+    return unless @shoutout[:user_id] == current_user.id
+
+    if Shoutout.exists?(digest: shoutout_params[:digest])
+      render json: { error: ERROR_1 }, status: :unprocessable_entity
+
+    elsif @shoutout.update(shoutout_params)
+
+      if @shoutout[:recipients].length.positive?
+        RecipientShoutout.where(shoutout_id: @shoutout.id).destroy_all
+        RecipientShoutout.create(records_recipients(@shoutout))
+      end
+
+      render json: @shoutout, status: :ok
+    else
+      render json: { error: @shoutout[:errors] }, status: :unprocessable_entity
+    end
```

Comment:
I see a lot of nested conditionals here.

Reply:
I refactored code to more clearer